cognite/client/_api/entity_matching.py

```diff
@@ -0,0 +1,220 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from requests import Response
+
+from cognite.client import utils
+from cognite.client._api_client import APIClient
+from cognite.client.data_classes._base import CogniteResource
+from cognite.client.data_classes.contextualization import (
+    ContextualizationJob,
+    EntityMatchingModel,
+    EntityMatchingModelList,
+    EntityMatchingModelUpdate,
+)
+from cognite.client.utils._auxiliary import convert_true_match
+
+
+class EntityMatchingAPI(APIClient):
+    _RESOURCE_PATH = EntityMatchingModel._RESOURCE_PATH
+    _LIST_CLASS = EntityMatchingModelList
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _run_job(self, job_path, json, status_path=None, headers=None, job_cls=None) -> ContextualizationJob:
+        job_cls = job_cls or ContextualizationJob
+        if status_path is None:
+            status_path = job_path + "/"
+        return job_cls._load_with_status(
+            self._post(self._RESOURCE_PATH + job_path, json=json, headers=headers).json(),
+            status_path=self._RESOURCE_PATH + status_path,
+            cognite_client=self._cognite_client,
+        )
+
+    def retrieve(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Optional[EntityMatchingModel]:
+        """Retrieve model
+
+        Args:
+            id: id of the model to retrieve.
+            external_id: external id of the model to retrieve.
+
+        Returns:
+            EntityMatchingModel: Model requested."""
+        utils._auxiliary.assert_exactly_one_of_id_or_external_id(id, external_id)
+        return self._retrieve_multiple(ids=id, external_ids=external_id, wrap_ids=True)
+
+    def retrieve_multiple(
+        self, ids: Optional[List[int]] = None, external_ids: Optional[List[str]] = None
+    ) -> EntityMatchingModelList:
+        """Retrieve models
+
+        Args:
+            ids: ids of the model to retrieve.
+            external_ids: external ids of the model to retrieve.
+
+        Returns:
+            EntityMatchingModelList: Models requested."""
+        utils._auxiliary.assert_type(ids, "id", [List], allow_none=True)
+        utils._auxiliary.assert_type(external_ids, "external_id", [List], allow_none=True)
+        return self._retrieve_multiple(ids=ids, external_ids=external_ids, wrap_ids=True)
+
+    def update(
+        self,
+        item: Union[
+            EntityMatchingModel, EntityMatchingModelUpdate, List[Union[EntityMatchingModel, EntityMatchingModelUpdate]]
+        ],
+    ) -> Union[EntityMatchingModel, List[EntityMatchingModel]]:
+        """Update model
+
+        Args:
+            item (Union[EntityMatchingModel,EntityMatchingModelUpdate,List[Union[EntityMatchingModel,EntityMatchingModelUpdate]]) : Model(s) to update
+        """
+        return self._update_multiple(items=item)
+
+    def list(self, filter: Dict = None, limit=100) -> EntityMatchingModelList:
```

Comment:
I suppose this is ok for a first iteration (i'm guessing this is lifted from an sdk for the playground api?) - but a part of going to beta should ideally be that we are confident that the api is stable enough that we can manage the cost of not having the filter be just a dict.